llama_index/vector_stores/pinecone.py

```diff
@@ -243,6 +251,43 @@ def __init__(
             batch_size=batch_size,
             remove_text_from_metadata=remove_text_from_metadata,
         )
+        pinecone = _import_pinecone()
+
+        # Cannot pass in str for pinecone_index
+        # Both clients >= 3.0.0 and < 3.0.0 should have pinecone.Index class
+        if not isinstance(pinecone_index, pinecone.Index):
+            raise ValueError(
+                f"`pinecone_index` should be an instance of pinecone.Index, "
+                f"got {type(pinecone_index)}"
+            )
+
+        self._pinecone_index = pinecone_index or self._initialize_pinecone_client(
+            api_key, index_name, environment, **kwargs
+        )
+
+    @staticmethod
+    def _initialize_pinecone_client(
```

Comment:
We could probably make these error messages more clear and have them focus on what the user needs to do to resolve their issue. 

We could also consider including information about the temporary version bifurcation.